test/e2e/operator_test.go

```diff
@@ -3533,22 +3555,21 @@ func TestUnsupportedConfigOverride(t *testing.T) {
 			if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
 				t.Fatalf("failed to get ingresscontroller deployment: %v", err)
 			}
-			if err := waitForDeploymentEnvVar(t, kclient, deployment, 30*time.Second, tt.env, ""); err != nil {
-				t.Fatalf("expected initial deployment not to set %s=true: %v", tt.env, err)
+			if err := waitForDeploymentEnvVar(t, kclient, deployment, 30*time.Second, tt.env, tt.defaultValue); err != nil {
+				t.Fatalf("expected initial deployment not to set %s=%s: %v", tt.env, tt.defaultValue, err)
```

Comment:
Even if the feature gate is enabled, we don't expect to see the max servers set to 1, the default?

Reply:
Correct, `ROUTER_MAX_DYNAMIC_SERVERS` is only set if the dynamic config manager is enabled: https://github.com/openshift/cluster-ingress-operator/blob/9532e7874711548509cb73f1a394f485ce80d816/pkg/operator/controller/ingress/deployment.go#L594-L611